chia/wallet/vc_wallet/vc_store.py

```diff
@@ -250,9 +250,16 @@ async def get_vc_record_by_coin_id(self, coin_id: bytes32) -> Optional[VCRecord]
 
     async def add_vc_proofs(self, vc_proofs: VCProofs) -> None:
         async with self.db_wrapper.writer_maybe_transaction() as conn:
-            await conn.execute(
-                "INSERT INTO vc_proofs VALUES(?, ?)", (vc_proofs.root().hex(), bytes(vc_proofs.as_program()))
-            )
+            root: bytes32 = vc_proofs.root()
+            exists_cursor = await conn.execute("SELECT EXISTS(SELECT 1 FROM vc_proofs WHERE root=?)", (root.hex(),))
```

Comment:
I think it's more reliable to use a context manager for `execute()` calls that returns a cursor. That way the cursor will be closed implicitly. Something like `async with conn.execute("...") as cursor:`

However, performing a lookup first and then decide whether to insert looks both expensive and risky, especially since this is just inside a *maybe*_transaction(). I would think it would be faster and more robust to just say:

```
INSERT OR IGNORE
```

or perhaps:

```
INSERT OR REPLACE
```

instead. I suspect that if the item exists, it's guaranteed to be identical to the one we're trying to insert. is that right? So `INSERT OR IGNORE` might be the way to go.

Reply:
You were correct INSERT OR IGNORE was just fine.  The change is much smaller now.